src/main/java/com/idrsolutions/microservice/BaseServlet.java

```diff
@@ -306,6 +307,25 @@ private static String sanitizeFileName(final String fileName) {
         return fileName.replaceAll("[\\\\/:\"*?<>| \\p{Cc}]", "_");
     }
 
+    /**
+     * Cap the filename to fit the specified file path limit.
+     * @param fileName name of the file uploaded
+     * @param context the ServletContext with access to the properties object
+     * @return String representing the filename beyond a given point removed to keep files within the file path limit
+     */
+    private static String capFileName(final String fileName, final ServletContext context) {
+        final Properties properties = (Properties) context.getAttribute(BaseServletContextListener.KEY_PROPERTIES);
+        final int filePathLimit = Integer.parseInt(properties.getProperty(BaseServletContextListener.KEY_PROPERTY_INTERNAL_FILENAME_LIMIT));
```

Comment:
Does capFileName need the ServletContext as a parameter? Surely the context.getAttribute line of code can be called before capFileName in outputFile?
```suggestion
    /**
     * Cap the filename to fit the specified file path limit.
     * @param fileName name of the file uploaded
     * @param properties a properties object with a internal filename limit value set
     * @return String representing the filename beyond a given point removed to keep files within the file path limit
     */
    private static String capFileName(final String fileName, final Properties properties) {
        final int filePathLimit = Integer.parseInt(properties.getProperty(BaseServletContextListener.KEY_PROPERTY_INTERNAL_FILENAME_LIMIT));
```